### PR TITLE
Update Dep: 2.13 changes

### PR DESCRIPTION
**Pull request by kii-dot**

### CODE REVIEW

File: common/src/main/scala/scalan/util/CollectionUtil.scala [hunk 1]
```diff
@@ -222,8 +221,8 @@ object CollectionUtil {
       res
     }
 
-    def filterCast[B:ClassTag](implicit cbf: CanBuildFrom[Source[A], B, Source[B]]): Source[B] = {
-      val b = cbf()
+    def filterCast[B:ClassTag](implicit cbf: Factory[B, Source[B]]): Source[B] = {
+      val b = cbf.newBuilder
```
Comment (aslesarenko) [inline common/src/main/scala/scalan/util/CollectionUtil.scala:225]: scala.collection.compat.BuildFrom should be used. I added the missing dependency in #807 and migrated this file. Please rollback changes and use as example in other places.

File: common/src/test/scala/sigmastate/VersionTesting.scala [hunk 2]
```diff
@@ -10,7 +10,7 @@ trait VersionTesting {
     * In the branch for v6.0 the new version 3 should be added so that the tests run for both.
     */
   protected val activatedVersions: Seq[Byte] =
-    (0 to VersionContext.MaxSupportedScriptVersion).map(_.toByte).toArray
+    (0 to VersionContext.MaxSupportedScriptVersion).map(_.toByte).toSeq
```
Comment (aslesarenko) [inline common/src/test/scala/sigmastate/VersionTesting.scala:13]: the right way is to use `.toArray[Byte]` as it just cannot infer the type for some reason.

File: core/src/main/scala/scalan/Base.scala [hunk 3]
```diff
@@ -467,7 +467,7 @@ abstract class Base { scalan: Scalan =>
       * and leave non-Ref items unchanged. */
     final def apply(xs: Seq[Any])(implicit o: Overloaded1): Seq[Any] = {
       val len = xs.length
-      if (len == 0) mutable.WrappedArray.empty
+      if (len == 0) Seq[Any]()
```
Comment (aslesarenko) [inline core/src/main/scala/scalan/Base.scala:470]: mutable.WrappedArray should be replaces with scala.collection.compat.immutable.ArraySeq. 
See recent commits in #807 where I replaces some of them.

Also, keep in mind that Sigma is almost entirely is part of hotspot code. 
Thus performance is critical and every allocation counts (basically every line is executed millions of times).
While migrating, introducing new allocations is not acceptable.
Please read https://github.com/ScorexFoundation/sigmastate-interpreter/blob/develop/docs/perf-style-guide.md for details of why Seq() is bad, and also for other performance patterns used in Sigma.

File: core/src/main/scala/scalan/compilation/GraphVizExport.scala [hunk 4]
```diff
@@ -131,7 +131,7 @@ trait GraphVizExport extends Base { self: Scalan =>
     case l: Long => s"${l}l"
     case arr: Array[_] => s"Array(${arr.toSeq.map(formatConst).mkString(", ")})"
     case seq: Seq[_] =>
-      s"${seq.stringPrefix}(${seq.map(formatConst).mkString(", ")})"
+      s"${seq.toString()}(${seq.map(formatConst).mkString(", ")})"
```
Comment (aslesarenko) [inline core/src/main/scala/scalan/compilation/GraphVizExport.scala:134]:
```suggestion
      s"Seq(${seq.map(formatConst).mkString(", ")})"
```

File: core/src/main/scala/scalan/staged/ProgramGraphs.scala [hunk 5]
```diff
@@ -31,7 +31,7 @@ trait ProgramGraphs extends AstGraphs { self: Scalan =>
 
     override def boundVars = Nil
     override def isIdentity: Boolean = false
-    override def freeVars = mutable.WrappedArray.empty[Sym]
+    override def freeVars = Seq[Sym]()
```
Comment (aslesarenko) [inline core/src/main/scala/scalan/staged/ProgramGraphs.scala:34]: See above comment about Seq()

File: sigmastate/src/test/scala/sigmastate/helpers/TestingHelpers.scala [hunk 6]
```diff
@@ -112,13 +112,13 @@ object TestingHelpers {
     * in our test cases
     */
   def createTransaction(outputCandidates: IndexedSeq[ErgoBoxCandidate]): ErgoLikeTransaction = {
-    new ErgoLikeTransaction(WrappedArray.empty, WrappedArray.empty, outputCandidates)
+    new ErgoLikeTransaction(IndexedSeq.empty, IndexedSeq.empty, outputCandidates)
```
Comment (aslesarenko) [inline sigmastate/src/test/scala/sigmastate/helpers/TestingHelpers.scala:115]: See above comment on how WrappedArray should be replaced.

**Review by aslesarenko — CHANGES_REQUESTED** [review]

File: core/src/main/scala/scalan/Base.scala (hunk 3 above)
Comment (k3ithl1m) [reply core/src/main/scala/scalan/Base.scala]: Should i change all occurence of Array to ArraySeq?

**Comment by aslesarenko** [comment]
Subsumed in #815

**Pull request closed without merging**